test/built-ins/RegExp/prototype/dotAll/this-val-invalid-obj.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2017 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Invoked on an object without an [[OriginalFlags]] internal slot
+esid: pending
+info: >
+    get RegExp.prototype.dotAll
+
+    1. Let R be the this value.
+    2. If Type(R) is not Object, throw a TypeError exception.
+    3. If R does not have an [[OriginalFlags]] internal slot, throw a TypeError
```

Comment:
3. If R does not have an [[OriginalFlags]] internal slot, then
  a. If SameValue(R, %RegExpPrototype%) is true, return undefined.
  b. Otherwise, throw a TypeError exception.